packages/alpinejs/src/magics/$model.js

```diff
@@ -0,0 +1,61 @@
+import { accessor } from '../interceptor'
+import { findClosest } from '../lifecycle'
+import { magic } from '../magics'
+import { reactive } from '../reactivity'
+
+magic('model', (el, { cleanup }) => {
+    let func = generateModelAccessor(el, cleanup)
+
+    Object.defineProperty(func, 'closest', { get() {
+        let func = generateModelAccessor(el.parentElement, cleanup)
+
+        func._x_modelAccessor = true
+
+        return accessor(func)
+    }, })
+
+    func._x_modelAccessor = true
+
+    return accessor(func)
+})
+
+function generateModelAccessor(el, cleanup) {
+    // Find the closest element with x-model on it, NOT including the current element...
+    let closestModelEl = findClosest(el, i => {
+        if (i._x_model) return true
+    })
+
+    // Instead of simply returning the get/set object, we'll create a wrapping function
+    // so that we have the option to add additional APIs without breaking anything...
+    let accessor = function (fallbackStateInitialValue) {
+        if (closestModelEl) return this
+
+        return fallbackStateInitialValue
+    }
+
+    let model = () => {
+        if (! closestModelEl) {
+            throw 'Cannot find an available x-model directive to reference from $model.'
+        }
+
+        return closestModelEl._x_model
+    }
+
+    accessor.get = () => {
```

Comment:
Could it be better to have these use the normal DOM event system? Would need a lot less upfront work, and less internal hackery. x-model will already be listening on something, no?

How does this currently handle when inputs exist inside a modeled tree? would a change event from the input not also trigger the model?

Reply:
I can't use the event system because that only covers the "set" story, not the "get" story. 

And you're right, x-modelable automatically removes event listeners, so I think I need some kind of solution for that problem here.

Options:
A) provide a dedicated API to remove listeners: `$model.removeListeners()` or something
B) automatically cleanup listeners when `$model.get()` is called for the first time
C) add a modifier to x-model like: `x-model.no-events` or some better name

I think B is my favorite - it's not a breaking change when introducing this feature, but would be after introducing the feature.

Thoughts?